gdb_helper

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+# Copyright (C) 2018 Luxoft
+
+# This is a help script for running gdb+gdbserver for remote debugging
+
+# To see the usage, run:
+# ./gdb_helper -h
+
+# To start gdbserver on target
+# ./gdb_helper -s
+
+# To transfer executable file from development machine to DEST on target and start gdb for remote debugging
+# ./gdb_helper -f exe_file -d DEST
+
+# To start gdbserver and attach to a running process with PID
+# ./gdb_helper -P PID
+
+# Then start gdb on dev machine for remote debugging
+# ./gdb_helper -g
+
+
+
+gdb_check() {
+    (env | grep GDB) > /dev/null
+    if [ "$?" != 0 ]
+    then
+        >&2 echo "No target GDB found, please source the target environment first from the SDK"
+        exit 1
+    fi
+}
+
+# Set default values
+target_ip=127.0.0.1
+ssh_port=1234
+gdb_port=3333
+dest=/tmp
+
+
+# Print usage information of the script
+usage() {
+    echo "usage: $0 [options]"
+    echo -e "[options] is any of the following:"
+    echo -e "  -f | --file            \tTransfer the executable file to the target and start gdb"
+    echo -e "  -d | --dest            \tThe destination folder on target where the file will be transferred"
+    echo -e "  -P | --PID             \tThe PID on target to be attached"
+    echo -e "  -i | --target_ip       \tThe target IP address. Defaults to 127.0.0.1"
+    echo -e "  -p | --ssh_port        \tSSH port to connect to QEMU. Defaults to 1234"
```

Comment:
`SSH port to connect to QEMU.` Is this script only applicable for an emulated target?

Reply:
Good catch!